microceph/common/network.go

```diff
@@ -0,0 +1,110 @@
+package common
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/canonical/lxd/shared/logger"
+)
+
+type NetworkIntf interface {
+	FindIpOnSubnet(subnet string) (string, error)
+	FindNetworkAddress(address string) (string, error)
+	IsIpOnSubnet(address string, subnet string) bool
+}
+
+type networkImpl struct{}
+
+// FindIpOnSubnet scans the host's network interfaces to check if an IP is available
+// for the provided subnet. It returns the found IP address or an empty string.
```

Comment:
Aiui the method will return the first unicast ipaddr it finds -- think that's fine but I'd suggest to note this in the doc string

Reply:
added.